### PR TITLE
Auto create employee / vendor

### PR DESCRIPTION
**Pull request by Sravanksk**

### CODE REVIEW

File: apps/netsuite/utils.py [hunk 1]
```diff
@@ -348,6 +416,68 @@ def sync_employees(self):
...
+                auto_map_employee_preference == 'EMPLOYEE_CODE' and employee.detail['employee_code']
+        ) else employee.detail['full_name']
+
+        employee = {
+            'entityId': employee_entity_id,
+            'firstName': employee.detail['full_name'].split(' ')[0],
+            'lastName': employee.detail['full_name'].split(' ')[-1]
+            if len(employee.detail['full_name'].split(' ')) > 1 else '',
+            'inheritIPRules': True,
+            'payFrequency': '_monthly',
```
Comment (ashwin1111) [inline apps/netsuite/utils.py:445]: what does this mean

File: apps/netsuite/utils.py [hunk 2]
```diff
@@ -348,6 +416,68 @@ def sync_employees(self):
...
+            'subsidiary': {
+                'name': None,
+                'internalId': subsidiary_mapping.internal_id,
+                'externalId': None,
+                'type': None
+            },
+            'workCalendar': {
+                "name": None,
+                "internalId": 1,
```
Comment (ashwin1111) [inline apps/netsuite/utils.py:454]: why is it hard coded as 1?

Comment (ashwin1111) [reply apps/netsuite/utils.py:454]: also, are both `'1' or 1` acceptable by netsuite?

File: apps/netsuite/utils.py [hunk 3]
```diff
@@ -348,6 +416,68 @@ def sync_employees(self):
...
+            'defaultExpenseReportCurrency': {
+                "internalId": currency.destination_id if currency else '1',
+                "externalId": None,
+                "type": "currency"
+            },
+            'externalId': employee.value
+        }
+        created_employee = self.connection.employees.post(employee)
+
+        employee_detail = self.connection.employees.get(created_employee['internalId'])
```
Comment (ashwin1111) [inline apps/netsuite/utils.py:467]: why do we need this call?
We already have `entityId, internalId, email` right even without this call right?

File: requirements.txt [hunk 4]
```diff
@@ -27,4 +27,4 @@ six==1.13.0
 sqlparse==0.3.0
 Unidecode==1.1.2
 urllib3==1.25.7
-wcwidth==0.1.8
```
Comment (ashwin1111) [inline requirements.txt:30]: don't we need to bump up netsuite sdk version?

File: apps/netsuite/utils.py [hunk 5]
```diff
@@ -312,6 +312,74 @@ def sync_vendors(self):
             vendor_attributes, self.workspace_id)
         return vendor_attributes
 
+    def post_vendor(self, vendor: ExpenseAttribute, auto_map_employee_preference: str, expense_group: ExpenseGroup):
```
Comment (ashwin1111) [inline apps/netsuite/utils.py:315]: most of the parts is same in `post_employees` and `post_vendors`, see if you can write a separate function for common parts

File: apps/netsuite/tasks.py [hunk 6]
```diff
@@ -70,19 +70,78 @@ def load_attachments(netsuite_connection: NetSuiteConnector, expense_id: str, ex
...
+
+        except NetSuiteRequestError as exception:
+            logger.exception({'error': exception})
+
+        except Exception:
+            error = traceback.format_exc()
+            detail = {
+                'error': error
+            }
+            logger.exception('Something unexpected happened workspace_id: %s %s', expense_group.workspace_id, detail)
```
Comment (ashwin1111) [inline apps/netsuite/tasks.py:123]: `Something unexpected happened during auto create employee / vendor workspace . . . ` would be easy for us to debug

Comment (ashwin1111) [reply apps/netsuite/tasks.py:123]: does is needs to be `logger.error` or `logger.exception`?

File: apps/netsuite/utils.py (hunk 1 above)
Comment (Sravanksk) [reply apps/netsuite/utils.py:445]: https://www.netsuite.com/help/helpcenter/en_US/srbrowser/Browser2019_1/schema/enum/employeepayfrequency.html?mode=package

please to check this

File: apps/netsuite/utils.py (hunk 2 above)
Comment (Sravanksk) [reply apps/netsuite/utils.py:454]: `workCalendar` is a required field - setting it to 1 by default, will check for any other possibilities here

File: apps/netsuite/tasks.py (hunk 6 above)
Comment (Sravanksk) [reply apps/netsuite/tasks.py:123]: this should be fine

File: apps/netsuite/utils.py (hunk 1 above)
Comment (Shwetabhk) [reply apps/netsuite/utils.py:443]: set to None

File: apps/netsuite/utils.py (hunk 2 above)
Comment (Shwetabhk) [reply apps/netsuite/utils.py:452]: set `internalId` to None

**Review by ashwin1111 — APPROVED** [review]

**Pull request merged**